lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -24,6 +44,47 @@ cudaDeviceInterface::~cudaDeviceInterface() {
     cleanup_memory();
 }
 
+void cudaDeviceInterface::set_device_clocks(unsigned int mem_clock, unsigned int core_clock,
+                                            unsigned int mem_count, unsigned int core_count) {
+
+    // Set default clocks to zero
+    uint32_t gpu_mem_clock = config.get_default<uint32_t>(unique_name, "gpu_mem_clock", 0);
+    uint32_t gpu_core_clock = config.get_default<uint32_t>(unique_name, "gpu_core_clock", 0);
+
+    uint32_t get_gpu_mem_clock, get_gpu_core_clock;
+
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_mem_clock, get_gpu_mem_clock);
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_core_clock, get_gpu_core_clock);
+
+    if (get_gpu_mem_clock != 0 & get_gpu_core_clock != 0) {
```

Comment:
```suggestion
    if (gpu_mem_clock != 0 && gpu_core_clock != 0) {
```
Use logical `&&` not bitwise `&` here.   Also I think you should be checking the clocks from the config, not the max clocks from above (which I don't think are needed).